### PR TITLE
Add pagination support for Sentry endpoints

### PR DESCRIPTION
**Pull request by olegpixel**
**Overview**
This PR introduces pagination support for Projects endpoint, enhancing the ability to retrieve data from Sentry's API efficiently.

Sentry Projects Endpoint - https://docs.sentry.io/api/organizations/list-an-organizations-projects/
Sentry Pagination Docs - https://docs.sentry.io/api/pagination/
Link Header - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link

**Changes Made**

Added a new FetchWithPagination method in the SentryClient to parse the Link header and extract the next page URL when available, aligning the code with Sentry's pagination guidelines.

Added a new GetAllProjects function to utilise the enhanced Fetch method, enabling it to fetch all projects across multiple pages as needed.

This change will address and resolve the issue described here - https://github.com/grafana/sentry-datasource/issues/31

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]
Hi @olegpixel, these changes LGTM! Just one question in the comments

File: pkg/sentry/sentry.go [hunk 1]
```diff
@@ -34,6 +35,44 @@ type SentryErrorResponse struct {
 	Detail string `json:"detail"`
 }
 
+func (sc *SentryClient) FetchWithPagination(path string, out interface{}) (string, error) {
+	fullURL := path
+	if !strings.HasPrefix(path, sc.BaseURL) {
+		fullURL = sc.BaseURL + path
+	}	
+	req, _ := http.NewRequest(http.MethodGet, fullURL, nil)
```
[the remark this answers is not in the document]
Comment (olegpixel) [reply pkg/sentry/sentry.go]: Hi @aangelisc, I've added the error check. Thank you!

**Comment by aangelisc** [comment]
Hey @olegpixel, the build is failing because of an unknown word `peterhellberg`. If you update the `cspell.config.json` file with this word the build will pass and we can merge 😊

**Comment by olegpixel** [comment]
Hi @aangelisc, I've updated `cspell.config.json` file. Thank you

**Review by aangelisc — APPROVED** [review]

**Comment by aangelisc** [comment]
Perfect, thanks again for the contribution @olegpixel!

**Pull request merged**